core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -36,23 +37,31 @@ export class ResultPanelComponent {
   private static readonly PRETTY_JSON_TEXT_LIMIT: number = 50000;
   private static readonly TABLE_COLUMN_TEXT_LIMIT: number = 1000;
 
+  //  show table when chartType is undefined, instead show visualization button
+  public chartType: string | undefined;
+  // record which operator is selected
+  public selectedOperatorID: string = '';
```

Comment:
I recommend remove this local variable, whenever this component needs to ask which operator is selected, it asks `JoinGraphWrapper`. 
This appraoch helps maintain a "single source of truth". If we maintain this variable, then we have the burdern of sync the local variable with global state.